modules/puppet_proxy/puppet_cache.rb

```diff
@@ -19,13 +19,13 @@ def scan_directory_with_cache directory, environment, scanner
         changed    = false
         manifest   = Dir.glob("#{directory}/*").map do |path|
           puppetmodule = File.basename(path)
-          mtime        = File.mtime(path)
+          mtime        = File.mtime(path).to_i.to_s
```

Comment:
Are you sure it'll serialise to a string?  JSON should support integers.

<pre>
irb(main):004:0> JSON.load({"mtime" => 123}.to_json)
=> {"mtime"=>123}
</pre>